src/python/pants/backend/python/util_rules/pex_test.py

```diff
@@ -535,3 +536,67 @@ def assert_description(
         use_repo_pex=True,
         expected="Extracting all requirements in lock.txt from repo.pex to build new.pex",
     )
+
+
+def test_error_on_invalid_lockfile_with_path(rule_runner: RuleRunner) -> None:
```

Comment:
Subjective style suggestion, feel free to disagree. It may be more readable to have this in a single test that puts `_run_pex_for_lockfile_test` as an inner helper function. These are all the same functionality, so I think makes sense to be in the same test as they're simply different edge cases of the same functionality. Kind of like pytest.parametrize.

Reply:
I thought about doing it that way, but my preferred behaviour is to match what `pytest.parametrize` does, which is to run every case and report every failure. Each one of these tests is running a completely different code path, and the behaviour has to be tested in a different way, otherwise I'd have done it with `parametrize`